internal/system/agent/executor.go

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package agent
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+)
+
+func runExec(serviceName string, operation string) (string, error) {
+	bytes, err := exec.Command(Configuration.ExecutorPath, serviceName, operation).CombinedOutput()
+	return string(bytes), err
+}
+
+func OperationViaExecutor(serviceName string, operation string) (string, error) {
```

Comment:
as I mentioned previously, I think having different interfaces that we type cast to to determine which operations a given implementation supports is a cleaner approach

Reply:
https://github.com/edgexfoundry/edgex-go/pull/1705#discussion_r319509366